containers/src/Kokkos_ScatterView.hpp

```diff
@@ -956,6 +966,7 @@ class ScatterView<DataType, Kokkos::LayoutRight, DeviceType, Op,
   ScatterView(std::string const& name, Dims... dims)
       : internal_view(view_alloc(WithoutInitializing, name),
                       unique_token.size(), dims...) {
+    // Think about this call
```

Comment:
These `reset()` calls in the constructors may be troublesome for always running on the default execution space. I could use feedback on how to handle them

Reply:
Would "accept an ExecSpace argument in the constructor, fence that" be an option? Something to the effect of

```c++
ScatterView(ExecSpace, name, dims) {
  reset(ExecSpace);
}
ScatterView(name, dims) : ScatterView(execution_space{}, name, dims) {}
```
?

Reply:
That seems straightforward, and I'd be happy to implement that.

My concern is more at the level of whether it's what's right for the use cases and the code base. I'll do a bit more examination of our application code to that end.

Reply:
I think that is going in the right direction. We need to check how the View Constructor (if at all) does this. Potentially we need to funnel this through the allocator arguments (i.e. view_alloc)).

Reply:
In our uses, the `ScatterView` is (re)initialized each time it's going to be used, along with the target `View`, as follows:

```c++
  if ( scatter_view.subview().extent(0) == 0 )
    scatter_view = decltype(scatter_view)(orig_view);
  scatter_view.reset();
  Kokkos::deep_copy(orig_view, 0.);
```

Everything happening there will need to be enqueued on a particular stream, and ideally not block the calling thread.